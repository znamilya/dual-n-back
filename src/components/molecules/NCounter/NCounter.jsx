import React, { PropTypes } from 'react';
import bemCl from 'bem-cl';

import './NCounter.styl';


const b = bemCl('n-counter');

class NCounter extends React.PureComponent {

    static propTypes = {
        value: PropTypes.number.isRequired,
        onChange: PropTypes.func.isRequired,
    };

    constructor(props) {
        super(props);

        this.handleMinusClick = this.handleMinusClick.bind(this);
        this.handlePlusClick = this.handlePlusClick.bind(this);
    }

    handleMinusClick() {
        this.props.onChange(this.props.value - 1);
    }

    handlePlusClick() {
        this.props.onChange(this.props.value + 1);
    }

    render() {
        return (
            <div className={b()}>
                <span className={b('control')} onClick={this.handleMinusClick}>-</span>
                <span className={b('value')}>{this.props.value}</span>
                <span className={b('control')} onClick={this.handlePlusClick}>+</span>
            </div>
        );
    }
}


export default NCounter;
