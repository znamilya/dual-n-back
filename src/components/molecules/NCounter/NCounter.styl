.n-counter
    display: flex
    align-items: center;


    &__value
        width: 0.6em
        margin: 0 10px

        text-align: center
        font-size: 30px
        line-height: 1


    &__control
        $size = .8em
        $bgColor = #aaa

        display: inline-block
        width: $size
        height: $size

        text-align: center
        font-size: 40px
        line-height: $size
        color: #fff

        user-select: none
        cursor: pointer
        border: 4px solid #fff
        background: $bgColor


        &:hover
            background-color: lighten($bgColor, 10%)
