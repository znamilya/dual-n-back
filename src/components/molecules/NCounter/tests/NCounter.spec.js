import React from 'react';

import NCounter from '../NCounter';


describe.only('<NCounter />', () => {
    it('should render', () => {
        const wrapper = shallow(
            <NCounter
                value={1}
                onChange={() => {}}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render n', () => {
        const n = 1;
        const wrapper = shallow(
            <NCounter
                value={n}
                onChange={() => {}}
            />
        );

        expect(+wrapper.find('.n-counter__value').text()).toEqual(n);
    });

    it('should increase n when user click on plus', () => {
        const n = 1;
        const onChange = jest.fn();
        const wrapper = shallow(
            <NCounter
                value={n}
                onChange={onChange}
            />
        );

        wrapper.find('.n-counter__control').first().simulate('click');

        expect(onChange).toHaveBeenCalledWith(n - 1);
    });

    it('should increase n when user click on plus', () => {
        const n = 1;
        const onChange = jest.fn();
        const wrapper = shallow(
            <NCounter
                value={n}
                onChange={onChange}
            />
        );

        wrapper.find('.n-counter__control').last().simulate('click');

        expect(onChange).toHaveBeenCalledWith(n + 1);
    });
});
