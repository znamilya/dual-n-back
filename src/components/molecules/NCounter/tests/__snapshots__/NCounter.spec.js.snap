exports[`<NCounter /> should render 1`] = `
<div
  className="n-counter">
  <span
    className="n-counter__control"
    onClick={[Function]}>
    -
  </span>
  <span
    className="n-counter__value">
    1
  </span>
  <span
    className="n-counter__control"
    onClick={[Function]}>
    +
  </span>
</div>
`;
